components/builder-worker/src/runner/util.rs

```diff
@@ -196,17 +196,9 @@ pub fn docker_exporter_spec(workspace: &Workspace) -> DockerExporterSpec {
             .expect("Project integrations must not be empty")
             .get_body(),
     ).expect("Project integrations body must be JSON");
-    let custom_tag = match opts.get("custom_tag") {
-        Some(val) => {
-            let val = val.as_str().expect("custom_tag value is a string");
-            if val.is_empty() {
-                None
-            } else {
-                Some(val.to_string())
-            }
-        }
-        None => None,
-    };
+    let custom_tag = get_optional_args(&opts, String::from("custom_tag"));
+    let registry_url = get_optional_args(&creds, String::from("registry_url"));
+    let registry_type = get_optional_args(&opts, String::from("integration"));
```

Comment:
I'm not sure if this needs to come from `opts` or `creds`. You're welcome future me.